src/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].tsx

```diff
@@ -171,6 +224,179 @@ export const parseSubjectPhaseSlug = (slug: string) => {
   };
 };
 
+export function formatCurriculumUnitsData(
```

Comment:
I think it'd be good to test `formatCurriculumUnitsData(...)` standalone with lots of different input data (battle test) as it seems quite important that we've got this part correct.